includes/class.llms.query.php

```diff
@@ -262,6 +264,29 @@ public function pre_get_posts( $query ) {
 
 	}
 
+	/**
+	 * Serve a 404 for certificates that are not viewable by the current user
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function maybe_404_certificate() {
+
+		if ( in_array( get_post_type(), array( 'llms_certificate', 'llms_my_certificate' ), true ) ) {
```

Comment:
```suggestion
		if ( is_singular( 'llms_my_certificate' ) ) {
```
I think this fits even better, also we don't need a check on the `llms_certificate` because it's not publicly queryable unless you can `manage_lifterlms`.

Reply:
mmm, yes, you are correct, good call.

Reply:
I'm still going to do a comparison against `get_post_type()` though because it's easier to mock for tests.